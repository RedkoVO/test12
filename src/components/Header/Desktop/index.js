import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import MenuNavigation from '../../MenuNavigation'
import Main from '../components/Main'

import styles from './styles'

const Header = props => {
  const { classes, onSubmit, balance, handleLogout, isDisabledButton } = props

  return (
    <header className={classes.root}>
      <MenuNavigation handleLogout={handleLogout} />
      <Main
        onSubmit={onSubmit}
        isDisabledButton={isDisabledButton}
        balance={balance && balance.shortBalance}
      />
    </header>
  )
}

Header.propTypes = {
  classes: PropTypes.object,
  onSubmit: PropTypes.func,
  handleLogout: PropTypes.func,
  balance: PropTypes.object,
  isDisabledButton: PropTypes.bool
}

export default withStyles(styles)(Header)