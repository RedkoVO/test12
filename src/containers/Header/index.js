import compose from 'recompose/compose'
import { connect } from 'react-redux'
import { withHandlers, lifecycle, pure } from 'recompose'

import { getBalance } from '../../redux/actions/balance'
import { getBalanceSelector } from '../../selectors/balance'

import Crypto from '../../crypto/crypto'
import { clearStorageForlogout } from '../../utils/localStorageUtils'

import AsyncHeaderDesktop from '../../components/Header/Desktop'

const mapStateToProps = state => ({
  balance: getBalanceSelector(state)
})

export default compose(
  connect(mapStateToProps),
  lifecycle({
    componentDidMount() {
      const isSecretKey = localStorage.getItem('secretKey')
      const getCryptoInfo = Crypto.account.accountFromSecret(isSecretKey)

      if (isSecretKey) {
        const data = {
          address: getCryptoInfo.address
        }
        this.props.dispatch(getBalance(data))
      }
    }
  }),
  withHandlers({
    handleLogout: ({ history }) => () => {
      clearStorageForlogout()
      history.push('/')
    }
  }),
  pure
)(AsyncHeaderDesktop)
