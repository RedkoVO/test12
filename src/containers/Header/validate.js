const validate = values => {
  const errors = {}

  if (!values.addressKey) {
    errors.addressKey = 'error.required'
  } else if (values.addressKey.length !== 64) {
    errors.addressKey = 'error.length'
  }

  if (!values.amount) {
    errors.amount = 'error.required'
  }

  return errors
}

export default validate