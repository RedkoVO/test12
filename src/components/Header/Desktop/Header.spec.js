import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Header from './'

it('renders main', () => {
  const tree = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <Header type={'main'} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders games', () => {
  const tree = renderer
    .create(<Header type={'games'} children={<div>Test</div>} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders catalog', () => {
  const tree = renderer.create(<Header type={'catalog'} />).toJSON()
  expect(tree).toMatchSnapshot()
})
