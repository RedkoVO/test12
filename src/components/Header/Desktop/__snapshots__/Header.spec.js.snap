// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders catalog 1`] = `
<div
  className="HeaderCatalog-root-28"
>
  <div
    className="HeaderCatalog-streamUserBlock-29"
  >
    <p
      className="HeaderCatalog-streamUserHello-30"
    >
       Hello, Ben A 
      <br />
      <span
        className="HeaderCatalog-streamUserVerification-32"
      >
        KYC VERIFICATED
      </span>
    </p>
    <div
      className="HeaderCatalog-streamUserPhoto-34"
    >
      <img
        alt=""
        src="user.png"
      />
    </div>
    <p
      className="HeaderCatalog-streamUserBallance-31"
    >
       764.23 DCB 
      <br />
      <span
        className="HeaderCatalog-streamUserBallDescr-33"
      >
        YOUR BALANCE
      </span>
    </p>
  </div>
</div>
`;

exports[`renders games 1`] = `
<div
  className="HeaderCatalog-root-21"
>
  <div
    className="HeaderCatalog-catalogUserBlock-22"
  >
    <p
      className="HeaderCatalog-catalogUserHello-23"
    >
       Hello, Ben A 
      <br />
      <span
        className="HeaderCatalog-catalogUserVerification-25"
      >
        KYC VERIFICATED
      </span>
    </p>
    <div
      className="HeaderCatalog-catalogUserPhoto-27"
    >
      <img
        alt=""
        src="user.png"
      />
    </div>
    <p
      className="HeaderCatalog-catalogUserBallance-24"
    >
       764.23 DCB 
      <br />
      <span
        className="HeaderCatalog-catalogUserBallDescr-26"
      >
        YOUR BALANCE
      </span>
    </p>
  </div>
  <div>
    Test
  </div>
</div>
`;

exports[`renders main 1`] = `
<header
  className="HeaderMain-root-1"
>
  <ul
    className="MenuNavigation-root-2"
  >
    <li
      className={undefined}
    >
      <div
        className="MenuNavigation-logout-11"
        onClick={[Function]}
      >
        Logout
      </div>
    </li>
    <li>
      <a
        className="MenuNavigation-headerMenuItem-3"
        href="/"
        onClick={[Function]}
      >
        <i
          className="MenuNavigation-menuIcon-4 MenuNavigation-dashboard-5"
        />
        <br />
         Dashboard
      </a>
    </li>
    <li>
      <a
        className="MenuNavigation-headerMenuItem-3"
        href="/"
        onClick={[Function]}
      >
        <i
          className="MenuNavigation-menuIcon-4 MenuNavigation-walletIcon-6"
        />
        <br />
         Wallet
      </a>
    </li>
    <li>
      <a
        className="MenuNavigation-headerMenuItem-3"
        href="/"
        onClick={[Function]}
      >
        <i
          className="MenuNavigation-menuIcon-4 MenuNavigation-shop-7"
        />
        <br />
         Shop
      </a>
    </li>
    <li>
      <a
        className="MenuNavigation-headerMenuItem-3"
        href="/games"
        onClick={[Function]}
      >
        <i
          className="MenuNavigation-menuIcon-4 MenuNavigation-games-8"
        />
        <br />
         Games
      </a>
    </li>
    <li>
      <a
        className="MenuNavigation-headerMenuItem-3"
        href="/"
        onClick={[Function]}
      >
        <i
          className="MenuNavigation-menuIcon-4 MenuNavigation-friends-9"
        />
        <br />
         Friends
      </a>
    </li>
    <li>
      <a
        className="MenuNavigation-headerMenuItem-3"
        href="/"
        onClick={[Function]}
      >
        <i
          className="MenuNavigation-menuIcon-4 MenuNavigation-settings-10"
        />
        <br />
         Settings
      </a>
    </li>
  </ul>
  <div
    className="UserBlock-root-12"
  >
    <div
      className="UserBlock-demoTitle-20"
    >
      DEMO VERSION
    </div>
    <div
      className="UserBlock-user-13"
    >
      <div
        className="UserBlock-userBlock-14"
      >
        <p
          className="UserBlock-userHello-15"
        >
           Hello, Ben A 
          <br />
          <span
            className="UserBlock-userVerification-16"
          >
            KYC VERIFICATED
          </span>
        </p>
        <div
          className="UserBlock-userPhoto-17"
        >
          <img
            alt=""
            src="user.png"
          />
        </div>
        <p
          className="UserBlock-userBallance-18"
        >
           dcb
          <br />
          <span
            className="UserBlock-userBallDescr-19"
          >
            YOUR BALANCE
          </span>
        </p>
      </div>
    </div>
  </div>
</header>
`;
